documentation/prplMesh-test-boardfarm.md

```diff
@@ -0,0 +1,85 @@
+This project can provide us with basic infrastructure for tests and save effort in terms of development and support.
+For details about integration process, design please see next sections.
+
+# Design
+
+By itself, [boardfarm](https://github.com/mattsm/boardfarm/) provides us with:
+1. Abstraction of test set-up
+1. Provides some basic classes of devices
+1. Some simple tests
+1. Handles connections to boards
+1. Provides test-suite organisation
+
+[boardfarm](https://github.com/mattsm/boardfarm/) already has some representation of some general devices (like Linux, OpenWRT device).
+But our test flows involve more components and require more info about devices\[1\].
+So we have to expand base classes with representation of `prplWRT` devices capable handling Agent, Controller and Sniffer.
+Also, some additional abstraction of easyMesh specific entities may ease future work(like CAPI, IEEE1905.1 message).
+
+Hierarchy of devices, entities their relation can be understood easier using typical test set-up scheme provided below:
+
+![test setup](/documentation/images/plantuml/boardfarm_prplMesh_test_setup.png)
+
+Please, note.
+In general case, it's expected that prplMesh executables are delivered to DUT (device under test) as OpenWRT `*.ipk`.
+DUT should be already burn with prplWRT or other supported platform.
+Thus, LAN connection required for flashing in uboot mode is missing from the scheme.
+
+In general, in order to start using [boardfarm](https://github.com/mattsm/boardfarm/) we have to create:
+
+1. Devices
+	1. prplWRT
+	1. dockerized dummy device
+	1. Wired sniffer
+	1. Wireless sniffer
+1. Additional entities
+	1. CAPI
+	1. prplMesh_controller
+	1. prplMesh_agent
+	1. prplMesh_radio_dummy
+	1. prplMesh_radio_hw
+
+Relations between classes and their hierarchy displayed below:
+
+
+![Classes scheme](/documentation/images/plantuml/boardfarm_prplMesh_classes.png)
+
+
+Please, note.
+That scheme is general and do not includes *all* methods and *fields*.
+If you are interested in them, please see source code.
+
+This additional infrastructure should be enough to start porting already existing tests from `test_flows.py`.
+
+## Integration
+
+[boardfarm](https://github.com/mattsm/boardfarm/) itself supports plugins\[2\].
+It allows to embed code to [boardfarm](https://github.com/mattsm/boardfarm/) which isn't present in main repository.
+All device classes, test specific entities and tests itself should be included like this.
+When they are stable enough, it's expected to push that to original [boardfarm](https://github.com/mattsm/boardfarm/) repository.
+
+General integration steps:
+
+1. Deliver boardfarm along with prplMesh.
+1. Create infrastructure.
+1. Move all tests from `test_flows.py`.
+1. Form boardfarm configuration for CI set up.
+1. Set up GitLab CI to trigger tests on PR merge.
```

Comment:
I don't think this is relevant in an architecture document that sits in the repository.

Instead, refer to /tests/boardfarm_plugins/README.md

Reply:
Ok. Removing it.